core/server/suite_test.go

```diff
@@ -74,3 +76,23 @@ func makeGRPCServer(cfg *rest.Config, t *testing.T) pb.CoreClient {
 
 	return pb.NewCoreClient(conn)
 }
+
+func withClientsPoolInterceptor(config *rest.Config, user *auth.UserPrincipal) grpc.ServerOption {
+	return grpc.UnaryInterceptor(func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
```

Comment:
Are we sure this runs? I have run into issues before where the middlewares don't run with our current implementation: https://github.com/grpc-ecosystem/grpc-gateway/issues/1043

Reply:
yes, if we comment that out the test breaks because the clients pool doesnt get injected

Reply:
What about when the server is actually running?

Reply:
I don't know if you pay attention to that's for test only since we use HTTP middleware for it.